multinet/api/tests/test_network.py

```diff
@@ -285,125 +269,106 @@ def test_network_rest_delete_unauthorized(workspace: Workspace, api_client: APIC
 
 
 @pytest.mark.django_db
-def test_network_rest_delete_forbidden(
-    workspace: Workspace,
-    user: User,
-    network_factory: NetworkFactory,
-    authenticated_api_client: APIClient,
-):
-    """Tests deleting a network on a workspace without sufficient permissions."""
-    workspace.set_user_permission(user, WorkspaceRoleChoice.READER)
-    network: Table = network_factory(workspace=workspace)
-    r = authenticated_api_client.delete(
-        f'/api/workspaces/{workspace.name}/networks/{network.name}/'
-    )
-
-    assert r.status_code == 403
-
-    # Assert relevant objects are not deleted
-    assert Network.objects.filter(name=network.name).exists()
-    assert workspace.get_arango_db().has_graph(network.name)
-
-
-@pytest.mark.django_db
-def test_network_rest_delete_no_access(workspace: Workspace, authenticated_api_client: APIClient):
-    """Test deleting a network from a workspace for which the user has no access at all."""
-    network = populated_network(workspace)
-    r = authenticated_api_client.delete(
-        f'/api/workspaces/{workspace.name}/networks/{network.name}/'
-    )
-    assert r.status_code == 404
-
-    # Assert relevant objects are not deleted
-    assert Network.objects.filter(name=network.name).exists()
-    assert workspace.get_arango_db().has_graph(network.name)
-
-
-@pytest.mark.django_db
-@pytest.mark.parametrize('permission', workspace_role_range())
+@pytest.mark.parametrize(
+    'permission,is_owner,status_code,success',
+    [
+        (None, False, 404, False),
+        (WorkspaceRoleChoice.READER, False, 200, True),
+        (WorkspaceRoleChoice.WRITER, False, 200, True),
+        (WorkspaceRoleChoice.MAINTAINER, False, 200, True),
+        (None, True, 200, True),
+    ],
+)
```

Comment:
An important distinction for tests that are similar to this one: If for a given test the mapping of `status_code` and `success` is one to one (e.g. here `404` is to `False` and `200` is to `True`), the I don't think we need to use the `status_code` parameterization for that test. We could just exclude it, and assert the status code directly if we need it. To me, this lowers complexity.

@naglepuff What do you think?

Reply:
I think for this test, and other tests that use `assert_limit_offset_results`, this suggestion is good. I like being able to assert the status code once instead of twice. I can't really do that here because this test uses a helper function with a hard-coded check for `200`. Since there's no way to do one `r.status_code == status_code` check here, I'm fine with getting rid of that parameter.